src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoDatabaseCustomNode.cs

```diff
@@ -14,14 +14,54 @@ namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel
     /// </summary>
     internal partial class DatabasesChildFactory : SmoChildFactoryBase
     {
+        private const string databaseUnavailableKey = "databaseUnavailable";
+
         public override string GetNodeStatus(object smoObject, SmoQueryContext smoContext)
         {
             return DatabasesCustomNodeHelper.GetStatus(smoObject, smoContext, CachedSmoProperties);
         }
+
+        public override Dictionary<string, object> GetExtraProperties(object smoObject, SmoQueryContext smoQueryContext)
+        {
+            var properties = new Dictionary<string, object>();
+            properties.Add(databaseUnavailableKey, DatabasesCustomNodeHelper.GetDatabaseIsUnavailable(smoObject, smoQueryContext, CachedSmoProperties));
```

Comment:
Minor: a nice pattern can be allowing an `AddExtraProperties` method that subclasses implement. This has 2 benefits:
- You always guarantee the dictionary is non-null
- If you have a node inheriting from its parent and both add properties it can call `super.AddExtraProperties` and get the combination instead of having to merge dictionaries

Reply:
Since this method is part of the factory, not the node itself, I'm not really sure what the best way to implement it is.

I've added a call to `base.GetExtraProperties` in the factory method implementation, but it looks like most node types don't use one of these factories.

Would you suggest changing the node code somehow?